token-lending/program/src/instruction.rs

```diff
@@ -283,6 +285,28 @@ pub enum LendingInstruction {
         /// The new owner
         new_owner: Pubkey,
     },
+    // 12
+    /// Make a flash loan.
+    ///
+    ///   0. `[writable]` Destination liquidity token account, minted by reserve liquidity mint.
+    ///   1. `[writable]` Reserve account.
+    ///   2. `[]` Lending market account.
+    ///   3. `[]` Derived lending market authority.
+    ///   4. `[]` Flash Loan Receiver Program Account, which should have a function (which we will
+    ///   call it `ExecuteOperation(amount: u64)` to mimic Aave flash loan) that has tag of 0.
+    ///   5. `[]` Flash Loan Receiver Program Derived Account
+    ///   6. `[]` Token program id
+    ///   7. `[writable]` Host fee receiver.
+    ///   8. `[writeable]` Flash loan fees receiver, must match init reserve.
+    /// ... a variable number of accounts that is needed for `executeOperation(amount: u64)`.
```

Comment:
Good point. One thing that I want to discuss is that whether we want to enforce that the `destination_liquidity_info` is owned by the `flash_loan_receiver` program. I think the plus side of not enforcing this is that it will make it more flexible.

For example if we transfer the fund to a `destination_liquidity_info` that is not owned by the `flash_loan_receiver` it could then call other program which returns the liquidity. Also given that we're not enforcing that the `flash_loan_receiver_program_derived_account` should be in the accounts expected, it will be hard to check if it's owned by the `flash_loan_receiver_program`.

Reply:
Yeah, I guess it's not strictly required to check. If the flash loan expects to use the funds and doesn't own it, it will just fail.